Common/Net/NetBuffer.cpp

```diff
@@ -76,7 +76,7 @@ bool Buffer::ReadAllWithProgress(int fd, int knownSize, RequestProgress *progres
 	int total = 0;
 	while (true) {
 		bool ready = false;
-		while (!ready && (!progress || !*progress->cancelled)) {
+		while (!ready && !progress) {
 			if (*progress->cancelled)
```

Comment:
When `progress` is null we shouldn't try to access it's member isn't?

Reply:
Hmm... Let me rethink this a little, I am still trying to understand RequestProgress.

Reply:
Seems like `progress->cancelled` is some sort of shared state? At least that's what I think it is... changed the code to reflect this.